cmd/argo/commands/cron/util.go

```diff
@@ -19,5 +18,9 @@ func GetNextRuntime(cwf *v1alpha1.CronWorkflow) (time.Time, error) {
 	if err != nil {
 		return time.Time{}, err
 	}
-	return cronSchedule.Next(time.Now().UTC()).Local(), nil
+	location, err := time.LoadLocation(cwf.Spec.Timezone)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return cronSchedule.Next(time.Now().In(location)).Local(), nil
```

Comment:
This will get the timezone of the Argo CLI, which is likely not the same as the timezone of the `workflow-controller` since clients usually run on local time and servers usually run on UTC, regardless of local time.

If this likely case is true, this change will provide incorrect information.

See recent discussion: https://github.com/argoproj/argo/issues/4614#issuecomment-735884722

Reply:
you are right, thanks for the information